modules/tinymce/src/plugins/table/test/ts/browser/SwitchTableSectionTest.ts

```diff
@@ -0,0 +1,274 @@
+  });
+
+  const basicContent = `<table>
+<tbody>
+<tr id="one">
+<td>text</td>
+</tr>
+<tr id="two">
+<td>text</td>
+</tr>
+</tbody>
+</table>`;
+
+  const theadExpected = `<table>
+<thead>
+<tr id="one">
+<td>text</td>
+</tr>
+</thead>
+<tbody>
+<tr id="two">
+<td>text</td>
+</tr>
+</tbody>
+</table>`;
+
+  const thsExpected = `<table>
+<tbody>
+<tr id="one">
+<th>text</th>
+</tr>
+<tr id="two">
+<td>text</td>
+</tr>
+</tbody>
+</table>`;
+
+  const reversedThsExpected = `<table>
+<tbody>
+<tr id="two">
+<td>text</td>
+</tr>
+<tr id="one">
+<th>text</th>
+</tr>
+</tbody>
+</table>`;
+
+  const bothExpected = `<table>
+<thead>
+<tr id="one">
+<th>text</th>
+</tr>
+</thead>
+<tbody>
+<tr id="two">
+<td>text</td>
+</tr>
+</tbody>
+</table>`;
+
+  const sNewHeaderSwitchTest = (tableHeaderType: string, startContent: string, expected: string) =>
+    Log.stepsAsStep('TINY-6007', `Switch new header, table_header_type = ${tableHeaderType}`, [
+      Chain.asStep({}, [
+        NamedChain.write('editor', McEditor.cFromSettings({
+          plugins: 'table',
+          theme: 'silver',
+          base_url: '/project/tinymce/js/tinymce',
+          table_header_type: tableHeaderType
+        })),
+        NamedChain.read('editor', ApiChains.cSetContent(startContent)),
+        NamedChain.direct('editor', Chain.mapper((editor: Editor) => editor.getBody()), 'body'),
+        NamedChain.read('editor', cSwitchSection('tr#one', 'thead')),
+        NamedChain.read('editor', ApiChains.cAssertContent(expected)),
+        NamedChain.read('editor', McEditor.cRemove)
+      ])
+    ]);
+
+  const existingTheadExpected = `<table>
+<thead>
+<tr id="one">
+<td>text</td>
+</tr>
+<tr id="two">
+<td>text</td>
+</tr>
+</thead>
+</table>`;
+
+  const existingThsExpected = `<table>
+<tbody>
+<tr id="one">
+<th>text</th>
+</tr>
+<tr id="two">
+<th>text</th>
+</tr>
+</tbody>
+</table>`;
+
+  const existingBothExpected = `<table>
+<thead>
+<tr id="one">
+<th>text</th>
+</tr>
+<tr id="two">
+<th>text</th>
+</tr>
+</thead>
+</table>`;
+
+  const thsAndTheadExpected = `<table>
+<thead>
+<tr id="two">
+<td>text</td>
+</tr>
+</thead>
+<tbody>
+<tr id="one">
+<th>text</th>
+</tr>
+</tbody>
+</table>`;
+
+  const theadAndThsExpected = `<table>
+<thead>
+<tr id="one">
+<td>text</td>
+</tr>
+</thead>
+<tbody>
+<tr id="two">
+<th>text</th>
+</tr>
+</tbody>
+</table>`;
+
+  const thsAndBothExpected = `<table>
+<thead>
+<tr id="two">
+<th>text</th>
+</tr>
+</thead>
+<tbody>
+<tr id="one">
+<th>text</th>
+</tr>
+</tbody>
+</table>`;
+
+  const theadAndBothExpected = `<table>
+<thead>
+<tr id="one">
+<td>text</td>
+</tr>
+<tr id="two">
+<th>text</th>
+</tr>
+</thead>
+</table>`;
+
+  const sExistingHeaderSwitchTest = (extraLabel: string, tableHeaderType: string, startContent: string, expected: string) =>
+    Log.stepsAsStep('TINY-6007', `Switch tbody to existing header, table_header_type = ${tableHeaderType}, ${extraLabel}`, [
```

Comment:
This could be simplified by using `Log.chainsAsStep`, as you don't need the `Chain.asStep` then. Other places in this test could also use this.